cmd/kube-proxy/app/server.go

```diff
@@ -546,19 +550,10 @@ func createClients(config componentbaseconfig.ClientConnectionConfiguration, mas
 }
 
 // Run runs the specified ProxyServer.  This should never exit (unless CleanupAndExit is set).
+// TODO: cannot return a nil error.
```

Comment:
// TODO: At the moment, Run() cannot return a nil error, otherwise it's caller will never exit. Update callers of Run to handle nil errors.

Reply:
Updated.
Thanks

Reply:
I have a little question. The func still return nil when success. 
```
func (s *ProxyServer) CleanUpAndExit() error {
	// remove iptables rules and exit
	if s.CleanupAndExit {
		encounteredError := userspace.CleanupLeftovers(s.IptInterface)
		encounteredError = iptables.CleanupLeftovers(s.IptInterface) || encounteredError
		encounteredError = ipvs.CleanupLeftovers(s.IpvsInterface, s.IptInterface, s.IpsetInterface, s.CleanupIPVS) || encounteredError
		if encounteredError {
			return errors.New("encountered an error while tearing down rules")
		}
	}
	return nil
}
```

Reply:
Where did you expect to handle nil error?
https://github.com/kubernetes/kubernetes/blob/1377108c084545e2279bd67eed56e54283572302/cmd/kube-proxy/app/server.go#L310-L311